src/hooks/posters.ts

```diff
@@ -81,7 +98,19 @@ export const usePosters = (posterHallId: string) => {
       return filteredPosterVenues.slice(0, displayedPostersCount);
 
     return fuseVenues
-      .search(searchQuery)
+      .search({
+        //@ts-ignore
+        $and: searchQuery
+          .trim()
+          .match(/(".*?"|[^"\s]+)+(?=\s*|\s*$)/g) // source: https://stackoverflow.com/a/16261693/1265472
```

Comment:
thanks for the buzz! Sadly for me it gives no "Show more details" (neither `Dismiss`) -- I guess because I am not a dev here. need to figure it out

Reply:
ok, first one is dealt with.  But I do not quite grasp on the 2nd one about `!`. Any clues @0xdevalias - e.g. is `!` having some meaning in JS regex? (it is not used in above one) or how to run similar checks locally so I could toy around quickly with some rough ideas?

Reply:
Looks like you got them both!

![image](https://user-images.githubusercontent.com/753891/119730427-0d855080-beb9-11eb-90d7-242d6a15f893.png)

![image](https://user-images.githubusercontent.com/753891/119730470-170eb880-beb9-11eb-8a28-a8eed18c9b96.png)

Reply:
> it runs in the client browser. Even if someone manages to hit it with entering many !s -- their fault and I do not think it would case any pain for server side (I could be proven wrong)

In general, yeah, that makes sense/agree. It's like a user-DoS that only impacts them if they do i (in this location). If it was triggerable from a chat message or something that would affect other users (even though clientside) that would be a bigger concern.

Reply:
(Fun sidenote, this is actually the first time I have seen the GitHub code scanning in action/pick something up, seems like a really neat feature! I doubt I ever would have noticed those Regex issues on my own; at least not without a lot more effort invested)

Reply:
Wait.. seems it still has issues with https://github.com/sparkletown/sparkle/pull/1443/commits/392007fd5b7cb862d150c656a14ac0dfc6fcf4c1

![image](https://user-images.githubusercontent.com/753891/119731078-d6636f00-beb9-11eb-9212-eb0590dfd88b.png)

Reply:
yeap, this one is still there but I can't see how/where `!` is relevant since it is not a part of the regex.  Does `.match` do something special? is there special treatment for `!` at all in JS regex?

Reply:
could be even some bug in the checker ;-)